packages/next/src/client/components/react-dev-overlay/pages/client.ts

```diff
@@ -1,90 +1,20 @@
 import * as Bus from './bus'
-import { parseStack } from '../utils/parse-stack'
-import { parseComponentStack } from '../utils/parse-component-stack'
-import {
-  hydrationErrorState,
-  storeHydrationErrorStateFromConsoleArgs,
-} from '../../errors/hydration-error-info'
 import {
   ACTION_BEFORE_REFRESH,
   ACTION_BUILD_ERROR,
   ACTION_BUILD_OK,
   ACTION_DEV_INDICATOR,
   ACTION_REFRESH,
   ACTION_STATIC_INDICATOR,
-  ACTION_UNHANDLED_ERROR,
-  ACTION_UNHANDLED_REJECTION,
   ACTION_VERSION_INFO,
 } from '../shared'
 import type { VersionInfo } from '../../../../server/dev/parse-version-info'
-import { attachHydrationErrorState } from '../../errors/attach-hydration-error-state'
 import type { DevIndicatorServerState } from '../../../../server/dev/dev-indicator-server-state'
+import { handleGlobalErrors } from '../../errors/use-error-handler'
+import { patchConsoleError } from '../../globals/intercept-console-error'
 
 let isRegistered = false
 
-function handleError(error: unknown) {
-  if (!error || !(error instanceof Error) || typeof error.stack !== 'string') {
-    // A non-error was thrown, we don't have anything to show. :-(
-    return
-  }
-
-  attachHydrationErrorState(error)
-
-  const componentStackTrace =
-    (error as any)._componentStack || hydrationErrorState.componentStack
-  const componentStackFrames =
-    typeof componentStackTrace === 'string'
-      ? parseComponentStack(componentStackTrace)
-      : undefined
-
-  // Skip ModuleBuildError and ModuleNotFoundError, as it will be sent through onBuildError callback.
-  // This is to avoid same error as different type showing up on client to cause flashing.
-  if (
-    error.name !== 'ModuleBuildError' &&
-    error.name !== 'ModuleNotFoundError'
-  ) {
-    Bus.emit({
-      type: ACTION_UNHANDLED_ERROR,
-      reason: error,
-      frames: parseStack(error.stack),
-      componentStackFrames,
-    })
-  }
```

Comment:
This is effectively removed but it's also not working at the moment as far as I can tell. Tracking in https://linear.app/vercel/issue/NDX-990